VelocitySDK/Operations/DetectOperations/VLTDrivingDetectOperation.m

```diff
@@ -0,0 +1,60 @@
+//
+//  VLTDrivingDetectOperation.m
+//  VelocitySDK
+//
+//  Created by Vytautas Galaunia on 16/08/2017.
+//  Copyright © 2017 Veloctity. All rights reserved.
+//
+
+#import "VLTDrivingDetectOperation.h"
+#import "Velocity.pb.h"
+#import "VLTSampleBuilder.h"
+
+static Float32 VLTSpeedThreshold = 20;
+
+@interface VLTDrivingDetectOperation ()
+
+@property (atomic, assign) Float32 speedThreshold;
+@property (atomic, assign) Float32 averageSpeed;
+@property (atomic, assign) BOOL isDriving;
+
+@end
+
+@implementation VLTDrivingDetectOperation
+
+- (void)processMotionRequest
+{
+    self.speedThreshold = VLTSpeedThreshold;
+
+    NSArray<VLTPBSensor *> *sensors = [self.motionRequest sensors];
+
+    NSUInteger gpsSensorIndex = [sensors indexOfObjectPassingTest:^BOOL(VLTPBSensor * sensor, NSUInteger idx, BOOL *stop) {
+        return sensor.type == VLTPBSensorTypeGps;
+    }];
+
+    if (gpsSensorIndex == NSNotFound) {
+        return;
+    }
+
+    VLTPBSensor *gpsSensor = sensors[gpsSensorIndex];
+
+
+    Float32 sum = 0;
+    for (VLTPBSample *sample in gpsSensor.samples) {
```

Comment:
Good catch. And also looked to the docs 
> Negative if speed is invalid.

But it doesn't say that location it self is invalid. Just the speed.